set.mm

```diff
@@ -49673,12 +49388,29 @@ That theorem bundles the theorems ( ` |- E. x ( x = y -> z e. x ) ` with
     BEAFZGQPQPBODBHABOOQOIJKLABMN $.
 
   ${
-    $d x y z $.
+    $d x y z w v $.
     $( Power set axiom: the powerclass of a set is a set.  Axiom 4 of
        [TakeutiZaring] p. 17.  (Contributed by NM, 30-Oct-2003.)  (Proof
        shortened by Andrew Salmon, 25-Jul-2011.)  Revised to prove ~ pwexg from
-       ~ vpwex .  (Revised by BJ, 10-Aug-2022.) $)
+       ~ vpwex .  (Revised by BJ, 10-Aug-2022.)  Reduce axiom usage.  (Revised
+       by Gino Giotto, 15-Sep-2024.) $)
     vpwex $p |- ~P x e. _V $=
+      ( vy vz vv vw cv cpw wel wi wal wb wex wceq ax-pow bm1.3ii dfcleq cab wsb
+      wcel wss df-pw dfss2 abbii eleq2i df-clab weq elequ2 imbi1d albidv sbievw
+      eqtri 3bitri bibi2i albii bitr2i exbii mpbi issetri ) BAFZGZCBHZDCHZDAHZI
+      ZDJZKZCJZBLBFZUTMZBLVEBCABCDNOVGVIBVIVACFZUTSZKZCJVGCVHUTPVLVFCVKVEVAVKVJ
+      DEHZVCIZDJZEQZSVOECRVEUTVPVJUTEFZUSTZEQVPEUSUAVRVOEDVQUSUBUCUKUDVOCEUEVOV
+      EECECUFZVNVDDVSVMVBVCECDUGUHUIUJULUMUNUOUPUQUR $.
+    $( $j usage 'vpwex' avoids 'ax-10' 'ax-11' 'ax-12'; $)
+  $}
+
+  ${
+    $d x y z $.
+    $( Obsolete version of ~ vpwex as of 15-Sep-2024.  (Contributed by NM,
+       30-Oct-2003.)  (Proof shortened by Andrew Salmon, 25-Jul-2011.)  Revised
+       to prove ~ pwexg from ~ vpwex .  (Revised by BJ, 10-Aug-2022.)
+       (Proof modification is discouraged.)  (New usage is discouraged.) $)
+    vpwexOLD $p |- ~P x e. _V $=
```

Comment:
I would prefer not to add ALT versions for ax-10, ax-11, ax-12, as many theorems can be proven without them and many proofs can be subject to revision. The technique for removing them is semi-automatic, similar but not quite as simple as for ax-13. As a result, most of these proofs can be shortened by factoring out and adding appropriate lemmas, which can be done later.

Reply:
Like @benjub , I am a little bit concerned that the proofs of many of (sometimes central, as in this case) theorems are enlarged heavily for the sake of avoiding axiom usage. This makes the proofs less understandable, which is against one (important) objective of Metamath (as described in the first chapter of the Metamath book by Norm). In principle, we have a dilemma with two contrary objectives of Metamath: to provide proofs as short as possible on the one side, and to use less or weaker axioms on the other side, whereas the second objective has precendence. This is fixed in the conventions for set.mm:
*  Minimizing axiom dependencies.
* Acceptable shorter proofs.

This dilemma was tried to be solved by using the ALT versions, see also the conventions:
* Alternate proofs (ALT): If a different proof is shorter or clearer but uses more or stronger axioms, we make that proof an "alternate" proof ...

To follow this convention is exactly what @benjub requested here.

If avoiding axiom usage in proofs is technical and can be "encapsulated" in lemmas, if I understand @GinoGiotto correctly, then these lemmas should be provided as soon as possible, at least before additional proofs are revised to avoid axiom usage of this kind. As long as these lemmas are not available, I would prefer to keep the original, short proofs as ALT versions, as proposed by @benjub .

Reply:
I provided a shortening, how does it look now?

The overall structure and approach remain the same: it uses [issetri](https://us.metamath.org/mpeuni/issetri.html), [bm1.3ii](https://us.metamath.org/mpeuni/bm1.3ii.html) and [axpow2](https://us.metamath.org/mpeuni/axpow2.html), just like the original. The only difference between the two is that I avoided using [abeq2](https://us.metamath.org/mpeuni/abeq2.html). I really don't think an ALT proof is worth here.

Reply:
OK, ~abeq2 (`( A = { x | ph } <-> A. x ( x e. A <-> ph ) )`) is expanded by the following sequence:

```
 4 dfcleq        $p |- ( w = { y | y C_ x } <-> A. z ( z e. w <-> z e. { y | y C_ x } ) )
 5 df-clab       $a |- ( z e. { y | y C_ x } <-> [ z / y ] y C_ x )
 6 sbss          $p |- ( [ z / y ] y C_ x <-> z C_ x )
 7 5,6 bitri     $p |- ( z e. { y | y C_ x } <-> z C_ x )
 8 7 bibi2i      $p |- ( ( z e. w <-> z e. { y | y C_ x } ) <-> ( z e. w <-> z C_ x ) )
 9 8 albii       $p |- ( A. z ( z e. w <-> z e. { y | y C_ x } ) <-> A. z ( z e. w <-> z C_ x ) )
10 4,9 bitr2i    $p |- ( A. z ( z e. w <-> z C_ x ) <-> w = { y | y C_ x } )
```
The only difference is that line 10 uses an additional setvar variable (x is replaced by y on the LHS. and by z on the RHS of ~abeq2, whereas A is replaced by w).
So maybe we should create a new theorem ~abeq2w (is the suffix w adeqate here?) for statement 10, proved by the sequence above. This can be used here (in the original proof of ~vpvex), replacing ~abeq2. Then ~abeq2 could be marked with "(New usage is discouraged.)". Currently, only a few theorems use ~abeq2:  abeq1 clabel rabid2 ru sbcabel dfss2OLD zfrep4 vpwex dmopab3 funimaexg bj-ru1

Reply:
The theorem you're referring to would appear as follows:

<pre>
  ${
    $d x y $.  $d y A $.  $d ph y $.  $d ps x $.
    abeq2w.1 $e |- ( x = y -> ( ph <-> ps ) ) $.
    $( Version of ~ abeq2 that requires less axioms.  (Contributed by GG and
       AV, 18-Sep-2024.) $)
    abeq2w $p |- ( A = { x | ph } <-> A. y ( y e. A <-> ps ) ) $=
      ( cab wceq cv wcel wb wal dfcleq wsb df-clab sbievw bitri bibi2i albii )
      EACGZHDIZEJZUATJZKZDLUBBKZDLDETMUDUEDUCBUBUCACDNBADCOABCDFPQRSQ $.
    $( $j usage 'abeq2w' avoids 'ax-8' 'ax-10' 'ax-11' 'ax-12' $)
  $}
</pre>

> (is the suffix w adeqate here?)

The difference between [abeq2](https://us.metamath.org/mpeuni/abeq2.html) and `abeq2w` looks similar to the difference between [sp](https://us.metamath.org/mpeuni/sp.html) and [spvv](https://us.metamath.org/mpeuni/spvv.html), the y variable does not appear in the end statement of [spvv](https://us.metamath.org/mpeuni/spvv.html) tho.

A nice additional property of `abeq2w` is that it does not require ax-8, contrary to [abeq2](https://us.metamath.org/mpeuni/abeq2.html).

Reply:
Yes, I would appreciate to add and use this theorem.

Reply:
Nice addition. Place it before df-clel

Comment:
> $( Version of ~ abeq2 using implicit substitution that requires fewer axioms.

(You can count axioms so it's "fewer", not "less".)